akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/internal/ReplicationImpl.scala

```diff
@@ -0,0 +1,216 @@
+                    val entityRef =
+                      entityRefFactory(destinationReplicaId.entityId).asInstanceOf[EntityRef[PublishedEvent]]
+                    if (log.isDebugEnabled) {
+                      log.debugN(
+                        "[{}] forwarding event originating on dc [{}] to [{}] (origin seq_nr [{}]): [{}]",
+                        projectionKey,
+                        replicatedEventMetadata.originReplica,
+                        destinationReplicaId.persistenceId.id,
+                        envelope.sequenceNr,
+                        replicatedEventMetadata.version)
+                    }
+                    val askResult = entityRef.ask[Done](replyTo =>
+                      PublishedEventImpl(
+                        replicationId.persistenceId,
+                        replicatedEventMetadata.originSequenceNr,
+                        envelope.event,
+                        envelope.timestamp,
+                        Some(new ReplicatedPublishedEventMetaData(
+                          replicatedEventMetadata.originReplica,
+                          replicatedEventMetadata.version)),
+                        Some(replyTo)))
+                    askResult.failed.foreach(error =>
+                      log.warn(
+                        s"Failing replication stream [$projectionName/$projectionKey] from [${remoteReplica.replicaId.id}], event pid [${envelope.persistenceId}], seq_nr [${envelope.sequenceNr}]",
+                        error))
+                    askResult
+
+                  case unexpected =>
+                    throw new IllegalArgumentException(
+                      s"Got unexpected type of event envelope metadata: ${unexpected.getClass} (pid [${envelope.persistenceId}], seq_nr [${envelope.sequenceNr}]" +
+                      ", is the remote entity really a Replicated Event Sourced Entity?")
+                }
+              } else {
+                // Events not originating on sending side already are filtered/have no payload and end up here
+                if (log.isDebugEnabled())
+                  log.debugN(
+                    "[{}] ignoring filtered event from replica [{}] (pid [{}], seq_nr [{}])",
+                    projectionKey,
+                    remoteReplica.replicaId,
+                    envelope.persistenceId,
+                    envelope.sequenceNr)
+                Future.successful(Done)
+              }
+          }))
+          .map(_ => Done)
+
+      val sourceProvider = EventSourcedProvider.eventsBySlices[AnyRef](
+        system,
+        eventsBySlicesQuery,
+        eventsBySlicesQuery.streamId,
+        sliceRange.min,
+        sliceRange.max)
+      ProjectionBehavior(settings.projectionProvider(projectionId, sourceProvider, replicationFlow, system))
+    }, shardedDaemonProcessSettings, None) // FIXME why doesn't Some(ProjectionBehavior.Stop) work
```

Comment:
The test hangs on shutdown if I enable sending stop. It kinda rings a bell but I couldn't find any previous issues around it. We made the projection always shut down in https://github.com/akka/akka-projection/pull/659 so it might be fine here, but still curious.

Reply:
We should debug that to understand, but could be a separate follow up. Some missing kill switch since we use flow based projection?

Reply:
Created an issue so we don't forget: https://github.com/akka/akka-projection/issues/777